packages/block/src/header.ts

```diff
@@ -189,12 +189,12 @@ export class BlockHeader {
           'Genesis parameters can only be set with a Common instance set to chainstart',
         )
       }
-      this.timestamp = this._common.genesis().timestamp
-      this.gasLimit = this._common.genesis().gasLimit
-      this.difficulty = this._common.genesis().difficulty
-      this.extraData = this._common.genesis().extraData
-      this.nonce = this._common.genesis().nonce
-      this.stateRoot = this._common.genesis().stateRoot
+      this.timestamp = toBuffer(this._common.genesis().timestamp || this.timestamp)
```

Comment:
Problem: `Common` lists some of these items as numbers while they should be Buffers. What do we do here? Edit Common to strictly enforce that genesis has the right types?

Reply:
why should they be buffers, wouldn't it be easier to store them as BN?

Reply:
Ah yeah I was about to respond to that. I will indeed convert them to BNs, that should fix this issue 👍 

For consistency we also have to check if `Tx` has any number fields and if they are also `BN`.

Reply:
great 👍 yes when working on the tx refactor I made sure to use the best domain types possible so anything that should be treated as a number should be `BN`.